create-i18n/index.test.ts

```diff
@@ -233,4 +236,237 @@ test('tracks double definition', () => {
   }, /I18n component double was defined multiple times/)
 })
 
+/* 
+----------- Partial loading testing -----------
+*/
+
+let partialGetCalls: object[] = []
+let partialResolveGet: (translations: ComponentsJSON[]) => void = () => {}
+let partialRequestsByLocale: Record<string, typeof partialResolveGet> = {}
+
+function partialGet(
+  code: string,
+  components: string[]
+): Promise<ComponentsJSON[]> {
+  partialGetCalls.push({ [code]: components })
+  return new Promise(resolve => {
+    partialRequestsByLocale[code] = resolve
+    partialResolveGet = resolve
+  })
+}
+
+async function partialGetResponse(
+  translations: ComponentsJSON[],
+  code?: string
+): Promise<void> {
+  if (code) {
+    partialRequestsByLocale[code](translations)
+  } else {
+    partialResolveGet(translations)
+  }
+}
+
+test.after.each(() => {
+  partialGetCalls = []
+})
+
+let locale = atom<'en' | 'ru' | 'fr' | 'de'>('ru')
+let i18n = createI18n(locale, { get: partialGet })
+let events: string[] = []
+
+let post = i18n('main/post', { title: 'Post' })
+let heading = i18n('main/heading', { title: 'Title' })
+let comment = i18n('main/comment', { title: 'Comment' })
+let message = i18n('chat/message', { title: 'Message' })
+let user = i18n('settings/user', { title: 'User' })
+let games = i18n('games', { title: 'Games' })
+
+let postUnbind: () => void
+let headingUnbind: () => void
+let commentUnbind: () => void
+
+test("after components mount shouldn't load translations with same prefix in componentName", async () => {
```

Comment:
Can we make test’s name shorter to fit 80 columns line?

Reply:
done